tests/support/butler.py

```diff
@@ -25,20 +25,34 @@ def size(self) -> int:
         return len(self.url) * 10
 
 
+class MockDatasetRef:
+    """Mock of a Butler DatasetRef."""
+
+    def __init__(self, uuid: UUID, dataset_type: str) -> None:
+        self.uuid = uuid
+        self.datasetType = self
+        self.name = dataset_type
```

Comment:
The name should be coming from `DatasetRef.datasetType.name`.  I think there is one level missing here

Reply:
Oh, sorry.  You set `self.datasetType = self`.  That looks like it would work